### ISSUE TITLE
Tab order login screen 'UX bug'

### ISSUE THREAD

**Issue by onno-timmerman** [wagtail/wagtail#5724]
### Issue Summary
It's not really a bug, but more a UX small annoying issue. 
When you come on wagtail login screen you fill in username hit key [TAB] fill in password hit key [TAB] and expect when you hit the enter key to be logged in. However due the tab order of the form you first pass the "forget password" option and enter that page.
The UI/UX experience is a bid off fro what you would expect

### Steps to Reproduce
Go to login screen login fast with only keyboard. 

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

#### Expected
The tab order should be
- login
- password
- submit
- forgot password

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project:yes 

### Technical details

* Python version: 3.7
* Django version: 2.2
* Wagtail version: 2.7

**Comment by jckpn** [comment]
I don't see an issue with the current tab order of the login page -- it follows the reading order (left-to-right, top-to-bottom) which is consistent with most websites.

![Screenshot 2019-12-12 at 16 46 30](https://user-images.githubusercontent.com/14837124/70732057-ab09b600-1cff-11ea-850e-d565ecf07bee.png)

It's worth remembering that tab order is an aspect of accessibility, so it's probably best to keep things as consistent and logical as possible, even if it's at the expense of us using it to speed through login forms :)

**Comment by onno-timmerman** [comment]
Hmmm, I'm not sure I agree this is logical. I this case I would say logic is from tab-order from top to bottom. https://ux.stackexchange.com/questions/48561/whats-the-best-tab-order-for-a-login-form

Login is done often, and you want to do it as quick as possible. So I would say different tab order :) ... However its not an important issue :)

**Comment by strindhaug** [comment]
I agree it's a bit surprising, though for me it's never an issue as i just type my username, tab, type password and then enter (without tabbing any more). It's only an issue if you tend to submit forms by tabbing to the submit button rather than just clicking enter inside the final form input.

But messing around with `tabindex` is almost always a real pain and tend to create more issues than it solves, so it's better just to move the forgot password link down by moving it after the submit button in the code so that it both visually and logically is after the submit button in the tab order.

We then probably should change the text on the link from "Forgotten it?" to "Forgot your password?" to make it more obvious. It's not as cute, but more standard and visible.

That would also make the forgot password link more easy to see as well, since you don't really expect it to be visually inside the textarea itself. (I've personally been confused for a few seconds wondering where the forgot password link was.)

**Comment by ababic** [comment]
Assigning to @kjartansverrisson :)

**Comment by kjartansverrisson** [comment]
I've got a PR in for this issue: https://github.com/wagtail/wagtail/pull/5775

**Comment by thibaudcolas** [comment]
Thank you @kjartansverrisson! This will be released in Wagtail 2.9. Thanks @onno-timmerman for taking the time to write such a detailed report to start with!
